bin/fetch-configlet

```diff
@@ -23,27 +36,45 @@ def get_os():
 
 
 def get_arch():
-    return '64bit' if sys.maxsize > 2**32 else '32bit'
+    return "64bit" if sys.maxsize > 2 ** 32 else "32bit"
 
 
 def fetch_configlet():
-    latest = "https://api.github.com/repos/exercism/configlet/releases/latest"
-    resp = requests.get(latest)
-    data = resp.json()
+    with urlopen(
+        "https://api.github.com/repos/exercism/configlet/releases/latest"
+    ) as request:
+        status = request.status
+        headers = dict(request.getheaders())
+        response = request.read()
+    if status >= 500:
+        print(f"Sleeping due to {status} response from API.")
+        time.sleep(5)
+        return 1
+    if status == 403:
+        wait_until = datetime.fromtimestamp(
+            int(headers["X-RateLimit-Reset"]), timezone.utc
+        )
+        delta = wait_until - datetime.now(timezone.utc)
+        seconds = delta.total_seconds()
+        wait = seconds + 5 if seconds > BACKOFF_TIME else BACKOFF_TIME
+        print(f"Rate limited, sleeping {wait} seconds.")
+        time.sleep(wait)
+        return 1
+    if status >= 400:
+        sys.exit(f"Received unexpected {status} response from API.")
+    data = json.loads(response.decode("utf-8"))
     version = data["tag_name"]
     machine_info = f"{get_os()}-{get_arch()}"
     name = f"configlet-{machine_info}.tgz"
     for asset in data["assets"]:
         if asset["name"] == name:
             print(f"Downloading configlet {version} for {machine_info}")
-            download_and_extract(asset["browser_download_url"])
-            return 0
+            return download_and_extract(asset["browser_download_url"])
     return 1
 
 
 def main():
-    max_retries = 5
-    for i in range(max_retries):
+    for i in range(MAX_RETRIES):
```

Comment:
If **i** is never actually used in the loop, maybe use `_`?

Reply:
It was like that originally, but I'll change it for style regardless.